core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -203,6 +206,10 @@ protected PositionFilterIterator(CloseableIterator<T> items, CloseableIterator<L
 
       @Override
       protected boolean shouldKeep(T row) {
+        return keepDeleteRows != filter(row);
+      }
+
+      private boolean filter(T row) {
```

Comment:
I think it would be better to have a subclass of this one that overrides `shouldKeep` instead:

```java
class PositionStreamDeletedRowSelector extends PositionStreamDeleteFilter {
  ...
  @Override
  protected boolean shouldKeep(T row) {
    return !super.shouldKeep(row);
  }
}
```